Sources/BitcoinKit/Core/Keys/PrivateKey.swift

```diff
@@ -31,98 +31,130 @@ import BitcoinKitPrivate
 #endif
 
 public struct PrivateKey {
+
     @available(*, deprecated, renamed: "data")
     public var raw: Data { return data }
+
     public let data: Data
+
     public let network: Network
-    public let isPublicKeyCompressed: Bool
+    public let shouldCompressPublicKey: Bool
+
+    /// The "designated" initializer of the `PrivateKey` struct, initialized from a scalar (represented as `Data` in lack of `BigInt` type). This scalar has to be within
+    /// the correct bounds, thus a validation check is done, why this initializer is throwing.
+    ///
+    /// - Parameter data: The private key scalar, represented as `Data` ( in lack of `BigInt` type), has to a value in range `0 < key < Secp256k1.orderOfCurve`
+    /// - Parameter network: Which network used
+    /// - Parameter shouldCompressPublicKey: Wether to compress public key or not.
+    public init(data: Data, network: Network = .testnet, shouldCompressPublicKey: Bool = true) throws {
```

Comment:
You can create new initializer and rename the properties, but you have to leave the original initializers and properties. And marks them as deprecated.

Please do your favor for the people already using this library.